src/infrasys/time_series_models.py

```diff
@@ -105,6 +106,7 @@ def check_data(cls, data) -> NDArray | pint.Quantity:  # Standarize what object
 
         if isinstance(data, pint.Quantity):
             if not isinstance(data.magnitude, np.ndarray):
+                # Why cant this use pint.Quantity instead of type(data)?
```

Comment:
I think we might need to run through one of the custom constructors.

Reply:
that makes sense

Reply:
Actually, I don't think that makes sense. Data is typed as np.array or pint.Quantity, and this branch is only considering data of type pint.Quantity, I don't think I saw any place where we are creating child classes from the data attribute class.